### DISCUSSION TITLE
Structuring many selectable variants

### DISCUSSION THREAD

**Discussion by TS1997** [pimcore/pimcore#9129] [Q&A / Support]
I'm a web developer and am currently working on a webshop for a company that sells consumables for large format printing as my first Pimcore project. They have products with up to 6+ variant attributes, e.g. color, width, height, thickness and a few yes/no options. All products doesn't have all options though, some might only have one or two options.

I'm now wondering how I should go about structuring the variants to get a scalable system that supports many different variant attributes without requiring a structure where every product have every possible variant attribute? I've looked through the documentation along with both the old apparel store and new car dealership demos. The apparel store seem to base the structure on a level-system where the products are structured as **Base -> Color -> Size** which doesn't seem like a good way in this case since there's a big variation in which attributes are used for variants for the different products.

This site will use one HTML select per attribute that is used as variant differentiation, e.g. one for color, one for thickness etc. I'd like to create a structure that allows me to not limit which order they select in. In other words I'd like it to be possible to select either color or thickness first and then the other if possible. I also want to be able to update the selectboxes based on previous selections. One example of that is that the width X doesn't have color Y for a specific product. That would require me to update the selectbox for color and disable Y if width X is selected and to disable width X if color Y was selected. 

What's the best practice for creating a structure that supports as much of this as possible? Any help, tips, tricks etc are greatly appreciated.

**Comment by passioneight** [comment]
You probably want to start off with the [ecommerce bundle](https://pimcore.com/docs/pimcore/current/Development_Documentation/E-Commerce_Framework/Configuration/PimcoreEcommerceFrameworkBundle_Configuration_Reference.html). 

It also sounds like you are looking for [objectbricks](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Object_Bricks.html).

Alternatively, leverage the [classification store](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Classification_Store.html).

As for the filtering: you'll have to implement it yourself or use a suitable bundle for this kind of task.
